src/org/thoughtcrime/securesms/NicknameHandler.java

```diff
@@ -0,0 +1,70 @@
+package org.thoughtcrime.securesms;
+
+import android.content.Context;
+
+import org.thoughtcrime.securesms.database.DatabaseFactory;
+import org.thoughtcrime.securesms.database.RecipientDatabase;
+import org.thoughtcrime.securesms.recipients.Recipient;
+
+
+public class NicknameHandler {
+    private Context context;
+    private RecipientDatabase recipientDatabase;
+
+    public NicknameHandler(Context context) {
+        this.context = context;
+    }
+
+    public NicknameHandler setContext(Context context) {
+        this.context = context;
+        return this;
+    }
+
+    /**
+     * This method allows the developer to set the underlining database
```

Comment:
Does this first sentence really say anything that cannot be seen instantly?

Reply:
Yes, it tells you that This method allows the developer to set the underlining database, it is important to note this point as long as we are using javadoc style

Reply:
It is literally called `SET database`

Reply:
yes but the rest of the comment is important in my opinion,

Reply:
yes but this method only used in testing

Reply:
I agree the second part is relevant, that's why I said first sentence in my original comment.

Reply:
I still believe that line of comment doesn't hurt to keep it. lets see what the other reviewers would say @shelbaz @erinb90 @Dan-Yoo @Radu-Raicea

Reply:
I think javadoc comments are pretty but I'm still conflicted since the code style guidelines say no comments... I guess keep them for now unless Dr. Shang or Jinfu says otherwise.

Reply:
@erinb90 it isn't about having that javadoc or not. My comment was ONLY about the first sentence.

Reply:
Oh yeah I agree the first sentence isn't needed.